### ISSUE TITLE
Pixelated Images at Output of Sampling Steps

### ISSUE THREAD

**Issue by ruchaa0112** [VSehwag/minimal-diffusion#2]
@VSehwag  - 
I trained the network on 1 channel by creating a custom data loader. 
The loss goes down as expected by 150 epochs. 
I also experimented with 200 epochs / 100 epochs.

On sampling 50 images using sampling steps = 250 / 500 /800 , I get some highly pixelated images amongst the 50 images. Following is the output of the pixelated image I got after using 800 sampling steps and training a model for 200 epochs. I used a batch size of 128.

Out of 50 sampled images , 10 images are highly pixelated. Others are smooth and match distribution of training data.

**Comment by ruchaa0112** [comment]
Hi @VSehwag - Just wanted to check if I was missing something during my training - 
I tried experimenting with a lot of hyperparameters - sampling steps / schedulers / diffusion steps / learning rate , but nothing helped till now.
Some generated images seem to be good , and some are highly pixelated.  There is no specific pattern in these images. 

Below are some pixelated images generated by network. 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/124633423/226710957-be8eb35a-0d44-4e10-9918-31090f16f2b8.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/124633423/226710977-10a71940-0a82-413e-9956-41b45ab2305f.png)
